lib/src/chip/chip.component.ts

```diff
@@ -0,0 +1,38 @@
+import { Component, Input, ViewEncapsulation, OnInit } from '@angular/core';
+import { anyToBoolean } from '../util';
+
+@Component({
+    selector: 'hc-chip',
+    template: `<div [ngClass]="chipType"><ng-content></ng-content></div>`,
+    styleUrls: ['./chip.component.scss'],
+    encapsulation: ViewEncapsulation.None
+})
+
+export class ChipComponent implements OnInit {
+
+    _action: boolean = false;
+    @Input() color: string = 'neutral';
+    chipType: string = 'chip';
+
+    constructor() {}
+
+    ngOnInit() {
+        this.chipType += ' ' + this.color;
+    }
+
+    @Input()
+    get action(): boolean {
+        return this._action;
+    }
+
+    set action(isAction) {
+        this._action = anyToBoolean(isAction);
+        if ( this._action ) {
+            this.chipType = 'chip close';
+        } else {
+            this.chipType = 'chip';
+        }
+
+        this.chipType += ' ' + this.color;
```

Comment:
Sorry I'm throwing alot at you here, with the above comments put together, I think you could do something like this:
```
@Component({
    selector: 'hc-chip',
    template: `<div [class]="color" [ngClass]="{'close': action, 'chip': true}"><ng-content></ng-content></div>`,
    styleUrls: ['./chip.component.scss'],
    encapsulation: ViewEncapsulation.None
})

export class ChipComponent {
    @Input() action: boolean = false;
    @Input() color: 'neutral' | 'yellow' | 'green' | 'red' = 'neutral';
}
```